Content.Server/GameObjects/Components/Nutrition/FoodComponent.cs

```diff
@@ -102,19 +161,28 @@ private bool TryUseFood(IEntity user, IEntity target)
 
             //We're empty. Become trash.
             var position = Owner.Transform.GridPosition;
-            Owner.Delete();
             var finisher = Owner.EntityManager.SpawnEntity(_trashPrototype, position);
-            if (user.TryGetComponent(out HandsComponent handsComponent) && finisher.TryGetComponent(out ItemComponent itemComponent))
+
+            // If the user is holding the item
+            if (user.TryGetComponent(out HandsComponent handsComponent) &&
+                handsComponent.IsHolding(Owner))
             {
-                if (handsComponent.CanPutInHand(itemComponent))
+                Owner.Delete();
+
+                // Put the trash in the user's hand
+                if (finisher.TryGetComponent(out ItemComponent item) &&
+                    handsComponent.CanPutInHand(item))
                 {
-                    handsComponent.PutInHand(itemComponent);
-                    return true;
+                    handsComponent.PutInHand(item);
+                    finisher.Transform.GridPosition = user.Transform.GridPosition;
```

Comment:
Is this really necessary? Something tells me `PutInHand` should handle that.

Reply:
It was a leftover from the user's hands being full and the trash being moved to the user's feet, which it doesn't do anymore. Changed it.